### ISSUE TITLE
File fetching broken since commit 0c1d2b9

### ISSUE THREAD

**Issue by ldaudet** [JakeChampion/fetch#1371]
Hi there!

Thanks a lot for the work on this library! I am using it in a react native app to run code like this: https://snack.expo.dev/@bacon/documentpicker-to-blob (line 24), to grab an image file that I will then upload to the backend.

This worked for the longest time, but after uploading to the latest version of expo, it also uploaded the version of react native, which in turned updated our version of fetch. The problem is that with the latest version of fetch, the fetch of file doesn't work anymore, as described here: https://github.com/facebook/react-native/issues/38625#issuecomment-1651430009

After looking at the code, I realized that the issue was actually do to the work on commit 0c1d2b9 where you added a check for the return code to be between certain values:
```
  if (this.status < 200 || this.status > 599) {
    throw new RangeError("Failed to construct 'Response': The status provided (0) is outside the range [200, 599].")
  }
```
When removing this check, everything works just fine with fetching the file, even with the lastest version of fetch, as the fetch of the a file return a status 0.

Would it be possible to either remove the check, or update the code to get a 200 status on successfully fetching a file?

Thanks a lot

**Comment by mercpls** [comment]
@ldaudet Still waiting on a response but there is a PR based on this suggestion now.

**Comment by meteorlxy** [comment]
According to [fetch spec - Schema fetch](https://fetch.spec.whatwg.org/#scheme-fetch):

> "file"
> For now, unfortunate as it is, file: [URLs](https://www.w3.org/TR/pub-manifest/#dfn-url) are left as an exercise for the reader.
>
> When in doubt, return a [network error](https://fetch.spec.whatwg.org/#concept-network-error).

I'm not sure what's the exact meaning of the spec - Should fetch supports file uri or not?

Any native speaker or spec expert can answer it? 🤔

**Comment by JakeChampion** [comment]
Fixed via #1375 👍

**Comment by mercpls** [comment]
> According to [fetch spec - Schema fetch](https://fetch.spec.whatwg.org/#scheme-fetch):
> 
> > "file"
> > For now, unfortunate as it is, file: [URLs](https://www.w3.org/TR/pub-manifest/#dfn-url) are left as an exercise for the reader.
> > When in doubt, return a [network error](https://fetch.spec.whatwg.org/#concept-network-error).
> 
> I'm not sure what's the exact meaning of the spec - Should fetch support file uri or not?
> 
> Any native speaker or spec expert could help answer it? 🤔

Not sure, but I recently saw in Expos docs (React Native framework/tooling) that they added a way to disable the polyfilling for fetch, but I haven't tested it personally.
<https://expo.github.io/router/docs/lab/runtime-location/#disable-polyfill>

I also included an alternate solution for native users in the meantime, so the existing broken code has a solution to be fixed.

I'm unsure if this should be something the native library owners need to update (using XMLHttpRequest internally when fetching a file) or if it's something that should be handled in this library (returning status for out of bound status errors on on file urls). I'd love if anyone has more insight.
